core/tests/protractor_utils/waitFor.js

```diff
@@ -207,6 +207,15 @@ var clientSideRedirection = async function(
   await browser.waitForAngularEnabled(true);
 };
 
+/**
+ * Wait for count of element.all Element Locater to be greater than 0
+ */
+var countToBeGreaterThanZero = async function(element, errorMessage) {
+  await browser.wait(async function() {
+    return await warningTextElements.count() > 0;
+  }, DEFAULT_WAIT_TIME_MSECS, errorMessage);
+};
+
```

Comment:
I think protractor doesn't wait for element.all locators to be visbile (ref https://github.com/angular/protractor/issues/2314). Hence designed another function which waits for the count of element.all locators to be greater than 0.

Reply:
The root issue in protractor seems to be angular/protractor#2917 where the issue is that if you say `expect(myElems.first().isPresent()).toBe(false)` then the test will raise an index bounds error on `myElems.first()` instead of having the test pass because no elements of `myElems` exist. However, that wouldn't cause issues with waitFor calls as angular/protractor#2314 claims.

The issues in the protractor repo also report an index out of bounds error, but that's not what's showing up in this flake. More generally, we follow this structure of waiting for `myElems.first()` a lot in the tests. If that structure caused index bounds errors, I would have expected us to see them more often.

Have you tried the `waitFor(myElems.first(), ...)` structure here and encountered issues with it?

Reply:
Done